WooCommerce/Classes/LocalAnnouncements/LocalAnnouncementsProvider.swift

```diff
@@ -0,0 +1,52 @@
+import Experiments
+import Foundation
+import Yosemite
+
+/// Provides the local announcements to be displayed on the dashboard tab.
+///
+final class LocalAnnouncementsProvider {
+    private let stores: StoresManager
+    private let analytics: Analytics
+    private let featureFlagService: FeatureFlagService
+    // The order of the announcements is based on the priority.
+    private let announcements: [LocalAnnouncement] = [.productDescriptionAI]
+
+    init(stores: StoresManager = ServiceLocator.stores,
+         analytics: Analytics = ServiceLocator.analytics,
+         featureFlagService: FeatureFlagService = ServiceLocator.featureFlagService) {
+        self.stores = stores
+        self.analytics = analytics
+        self.featureFlagService = featureFlagService
+    }
+
+    func loadAnnouncement() async -> LocalAnnouncementViewModel? {
+        guard featureFlagService.isFeatureFlagEnabled(.productDescriptionAIFromStoreOnboarding) else {
+            return nil
+        }
+        for announcement in announcements {
+            guard isEligible(announcement: announcement), await isVisible(announcement: announcement) else {
+                continue
+            }
+            return .init(announcement: announcement, stores: stores, analytics: analytics)
+        }
+        return nil
```

Comment:
❓ I'm not sure I understand - is this method to load the details of only the first announcement? Can we make this clearer by mapping the `announcements` array and get the first item?

```suggestion
        let details = announcements.map {
            guard isEligible(announcement: announcement), await isVisible(announcement: announcement) else {
                continue
            }
            return .init(announcement: announcement, stores: stores, analytics: analytics)
        }
        return details.first
```

It would be great to comment on the method to clarify the intention. Would `loadAnnouncementToDisplay` be more precise for the method name?

Reply:
There could be multiple announcements in the app, but we only want to display one at a time (whenever dashboard's `syncAnnouncements` is called and no JITM modal is displayed). This function goes through the list of announcements in the order of priority, and returns the first one that's eligible and visible (the user hasn't dismissed). `map` wasn't used for the array of announcements because an announcement at a lower index can be returned, and it can save some unnecessary async checks. The length of the array is expected to be manageable and maintained locally by devs, so checking it sequentially is not likely to cause a performance issue.

I'll update the comment for the function to be more clear.

Reply:
Comment added in https://github.com/woocommerce/woocommerce-ios/pull/10118/commits/f58ff14e33b00134ed700338f11ddb080f3e4dde.